experiment_executor.py

```diff
@@ -229,45 +234,27 @@ def save_delays_and_imbalance(self, experiment):
-            y = self.imbalance_df[self.imbalance_df['algorithm'] == group]['imbalance_percentage'].tolist()
-            plt.plot(x, y, label=group, linewidth=2)
-
-        path = "experiments/" + experiment + "/imbalance" + "_" + getCurrentDateTime()
-        plt.legend(loc="upper right")
-        # TODO: Change label to more readable representation
-        plt.xlabel(experiment)
-        plt.ylabel("Percentage value of imbalance")
-        plt.savefig(path + ".png")
-
-        plt.clf()
-        for group in self.delays_df['algorithm'].unique():
-            x = self.delays_df[self.delays_df['algorithm'] == group][experiment].tolist()
-            y = self.delays_df[self.delays_df['algorithm'] == group]['delay_percentage'].tolist()
-            plt.plot(x, y, label=group, linewidth=2)
-
-        path = "experiments/" + experiment + "/delays" + experiment + "_" + getCurrentDateTime()
-        plt.legend(loc="upper right")
-        # TODO: Change label to more readable representation
-        plt.xlabel(experiment)
-        plt.ylabel("Percentage value of total delay")
-        plt.savefig(path + ".png")
-
-        plt.clf()
-        for group in self.estimated_delays['algorithm'].unique():
-            x = self.estimated_delays[self.estimated_delays['algorithm'] == group][experiment].tolist()
-            y = self.estimated_delays[self.estimated_delays['algorithm'] == group]['delay_percentage'].tolist()
-            plt.plot(x, y, label=group, linewidth=2)
-
-        path = "experiments/" + experiment + "/estimated_delays_" + experiment + "_" + getCurrentDateTime()
-        plt.legend(loc="upper right")
-        # TODO: Change label to more readable representation
-        plt.xlabel(experiment)
-        plt.ylabel("Percentage value of estimated total delay")
-        plt.savefig(path + ".png")
+        plot_params = {
+            "dataframe": [self.imbalance_df, self.delays_df, self.estimated_delays],
+            "df_column": ["imbalance_percentage", "delay_percentage", "delay_percentage"],
+            "path_folder": ["/imbalance_", "/delays_", "/estimated_delays_"],
+            "plot_y_label": ["Percentage value of imbalance", "Percentage value of total delay", "Percentage value of total delay"]
+        }
+
+        for index in range(3):
+            plt.clf()
+            for group in plot_params["dataframe"][index]['algorithm'].unique():
+                x = plot_params["dataframe"][index][plot_params["dataframe"][index]['algorithm'] == group][experiment].tolist()
+                y = plot_params["dataframe"][index][plot_params["dataframe"][index]['algorithm'] == group][plot_params["df_column"][index]].tolist()
+                plt.plot(x, y, label=group, linewidth=2)
+            path = "experiments/" + experiment + plot_params["path_folder"][index] + experiment + "_" + getCurrentDateTime()
+            plt.legend(loc="upper right")
+            # TODO: Change label to more readable representation
```

Comment:
maybe this comment is not needed aymore?

Reply:
Deleted